packages/botfuel-dialog/src/dialog-manager.js

```diff
@@ -108,15 +112,16 @@ class DialogManager extends Resolver {
       newDialog = {
         name: classificationResults[0].name,
         data: classificationResults[0].isQnA()
-          ? { answers: classificationResults[0].answers }
+          ? { answers: (classificationResults[0].answers: any) } // TODO refactor (law of Demeter)
```

Comment:
`ClassificationResult.ansers` is already of type `?QnaAnswers` (= `QnaAnswers | void`).

The issue is with `DialogDataData.answers`:
```
export type DialogDataData = {
  classificationResults?: ClassificationResult[],
  messageEntities?: MessageEntities,
  answers?: QnaAnswers,
  url?: string,
};
```

The member `answers` is either not present or of type `QnaAnswers`.

So the following assignment is illegal because `answer` could be present but with type `void`:
```
      newDialog = {
        name: classificationResults[0].name,
        data: classificationResults[0].isQnA()
          ? { answers: (classificationResults[0].answers: any) } // TODO refactor (law of Demeter)
          : { messageEntities },
      };
```

However we could explicitely allow this case: `  answers?: ?QnaAnswers,`

Reply:
I think the problem here is that if you define ClassificationResult.ansers type ?QnaAnswers, it can be null as well (  https://flow.org/en/docs/types/maybe/ ). And since in data answers is optional so it can only be undefined or QnaAnswers, this is not compatible and that's why you need to put :any. But if we changed to QnaAnswers | void ==> or QnaAnswers, or undefined then removing :any works.

Reply:
You're right!